Sources/NIOAsyncAwaitDemo/main.swift

```diff
@@ -57,6 +57,9 @@ func main() async {
         print(", response:", String(buffer: response2.body ?? ByteBuffer()))
 
         try await channel.close()
+
+        try await group.shutdownGracefully()
```

Comment:
This patch isn't quite right as we now shut the loop down twice. Can you move the existing event loop construction out of the top-level scope and into this async function instead?

Reply:
Yes, I noticed this.

I wasn't able able to use the `defer` shutdown with the `async`, and because the comments said it was safe to call shutdown twice, I thought the safest thing to do was to leave the top-level shutdown.

I'll move the group lifecycle into the `async` function, though as it's probably the right thing to do.

Reply:
Yeah, you can't `await` in a `defer`. In this case, a `do...catch` is probably easier.

Reply:
Lemme add the shutdown to the `catch`.